### ISSUE TITLE
java.lang.NoClassDefFoundError: com/sun/mirror/apt/AnnotationProcessorFactory

### ISSUE THREAD

**Issue by dpisarenko** [jcabi/jcabi-ssl-maven-plugin#27]
**Steps to reproduce:**
  1. `git pull origin master`
  1. `mvn clean install -Pqulice`

**Result**
```
java.lang.NoClassDefFoundError: com/sun/mirror/apt/AnnotationProcessorFactory
        at java.lang.Class.getDeclaredConstructors0(Native Method)
        at java.lang.Class.privateGetDeclaredConstructors(Class.java:2658)
        at java.lang.Class.getDeclaredConstructors(Class.java:2007)
        at com.google.inject.spi.InjectionPoint.forConstructorOf(InjectionPoint.
java:245)
        at com.google.inject.internal.ConstructorBindingImpl.create(ConstructorB
indingImpl.java:99)
        at com.google.inject.internal.InjectorImpl.createUninitializedBinding(In
jectorImpl.java:653)
        at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(Injec
torImpl.java:863)
        at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecurs
ive(InjectorImpl.java:790)
        at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(Injector
Impl.java:278)
        at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImp
l.java:210)
        at com.google.inject.internal.InjectorImpl.getProviderOrThrow(InjectorIm
pl.java:986)
        at com.google.inject.internal.InjectorImpl.getProvider(InjectorImpl.java
:1019)
        at com.google.inject.internal.InjectorImpl.getProvider(InjectorImpl.java
:982)
        at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java
:1032)
        at org.eclipse.sisu.space.AbstractDeferredClass.get(AbstractDeferredClas
s.java:48)
        at com.google.inject.internal.ProviderInternalFactory.provision(Provider
InternalFactory.java:86)
        at com.google.inject.internal.InternalFactoryToInitializableAdapter.prov
ision(InternalFactoryToInitializableAdapter.java:55)
        at com.google.inject.internal.ProviderInternalFactory$1.call(ProviderInt
ernalFactory.java:70)
        at com.google.inject.internal.ProvisionListenerStackCallback$Provision.p
rovision(ProvisionListenerStackCallback.java:100)
        at org.eclipse.sisu.plexus.PlexusLifecycleManager.onProvision(PlexusLife
cycleManager.java:133)
        at com.google.inject.internal.ProvisionListenerStackCallback$Provision.p
rovision(ProvisionListenerStackCallback.java:109)
        at com.google.inject.internal.ProvisionListenerStackCallback.provision(P
rovisionListenerStackCallback.java:55)
        at com.google.inject.internal.ProviderInternalFactory.circularGet(Provid
erInternalFactory.java:68)
        at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(
InternalFactoryToInitializableAdapter.java:47)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(Pr
oviderToInternalFactoryAdapter.java:46)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.ja
va:1054)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(Provi
derToInternalFactoryAdapter.java:40)
        at com.google.inject.Scopes$1$1.get(Scopes.java:59)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(Inter
nalFactoryToProviderAdapter.java:41)
        at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:99
7)
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.ja
va:1047)
        at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:993)
        at org.eclipse.sisu.inject.LazyBeanEntry.getValue(LazyBeanEntry.java:82)

        at org.eclipse.sisu.plexus.LazyPlexusBean.getValue(LazyPlexusBean.java:5
1)
        at java.util.AbstractMap.get(AbstractMap.java:187)
        at org.apache.maven.tools.plugin.scanner.DefaultMojoScanner.populatePlug
inDescriptor(DefaultMojoScanner.java:99)
        at org.apache.maven.plugin.plugin.AbstractGeneratorMojo.execute(Abstract
GeneratorMojo.java:233)
        at org.apache.maven.plugin.plugin.HelpGeneratorMojo.execute(HelpGenerato
rMojo.java:88)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(Default
BuildPluginManager.java:132)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor
.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor
.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor
.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProje
ct(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProje
ct(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThre
adedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(Lifecycl
eStarter.java:120)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.
java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAcces
sorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Laun
cher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.jav
a:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(La
uncher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:
356)
Caused by: java.lang.ClassNotFoundException: com.sun.mirror.apt.AnnotationProces
sorFactory
        at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(
SelfFirstStrategy.java:50)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadCl
ass(ClassRealm.java:259)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm
.java:235)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm
.java:227)
        ... 59 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.443 s
[INFO] Finished at: 2015-01-28T15:46:17+04:00
[INFO] Final Memory: 12M/30M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-plugin-plugin:3.2:
helpmojo (generated-helpmojo) on project jcabi-ssl-maven-plugin: The API of the
mojo scanner is not compatible with this plugin version. Please check the plugin
 dependencies configured in the POM and ensure the versions match. com/sun/mirro
r/apt/AnnotationProcessorFactory: com.sun.mirror.apt.AnnotationProcessorFactory
-> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e swit
ch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please rea
d the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionE
xception
```
**Environment:**
```
$ mvn -version
Apache Maven 3.2.2 (45f7c06d68e745d05611f7fd14efb6594181933e; 2014-06-17T17:51:4
2+04:00)
Maven home: c:\Program Files\apache-maven-3.2.2
Java version: 1.8.0_20, vendor: Oracle Corporation
Java home: c:\Program Files\Java\jdk1.8.0_20\jre
Default locale: de_AT, platform encoding: Cp1252
OS name: "windows 7", version: "6.1", arch: "x86", family: "dos"
```

**Comment by Suseika** [comment]
I'm experiencing the same problem. Here is the solution: [java.lang.NoClassDefFoundError: com/sun/mirror/apt/AnnotationProcessorFactory when execute Enunciate ant task](http://stackoverflow.com/questions/29402846/java-lang-noclassdeffounderror-com-sun-mirror-apt-annotationprocessorfactory-wh). 

Basically, that mojo plugin doesn't work with Java 8 JDK, so you just should use Java 7 JDK and it will build fine. 

I think that we should make that explicit in the build by using maven-enforcer-plugin to force Java 7.

**Comment by dmarkov** [comment]
@yegor256 please pay attention to this issue

**Comment by dmarkov** [comment]
@suseika it's your task now

**Comment by Suseika** [comment]
@dpisarenko I fixed this in #34, please see and close the issue if you're fine with the implementation.

**Comment by Suseika** [comment]
@dmarkov I'm done here.

**Comment by Suseika** [comment]
@dmarkov I closed this issue with #34, please send a payment or explain what else needs to be done here.

**Comment by dmarkov** [comment]
@suseika thanks, I just added **30 mins** to your account, payment `AP-61K49397GX2707227`, 98 hours and 39 mins spent. +30 to your rating, your total score is [+188](http://www.netbout.com/b/35381?open=rating)
